src/MSBuild/XMake.cs

```diff
@@ -956,16 +956,15 @@ string[] commandLine
                 s_buildComplete.Set();
                 Console.CancelKeyPress -= cancelHandler;
 
-                // Wait for any pending cancel, so that we get any remaining messages
-                s_cancelComplete.WaitOne();
-
 #if FEATURE_GET_COMMANDLINE
                 MSBuildEventSource.Log.MSBuildExeStop(commandLine);
 #else
                 if (MSBuildEventSource.Log.IsEnabled()) {
                     MSBuildEventSource.Log.MSBuildExeStop(string.Join(" ", commandLine));
                 }
 #endif
+                // Wait for any pending cancel, so that we get any remaining messages
+                s_cancelComplete.WaitOne();
```

Comment:
Seems odd to put this wait outside the timing; what was the motivation here?

Reply:
Good catch. I will put it back. Motivation was wrong. I was trying to make sure it is logged my logic was flawed - at best.